### PR TITLE
Use the PennyLane Sphinx Theme

### PR DESCRIPTION
**Pull request by josh146**
**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -170,12 +170,12 @@
...
-html_sidebars = {
-    "**": [
-        "searchbox.html",
-        "globaltoc.html",
-    ]
-}
+# html_sidebars = {
+#     "**": [
+#         "searchbox.html",
+#         "globaltoc.html",
```
Comment (antalszava) [inline doc/conf.py:176]: Minor: should we delete there?
```suggestion
```

**Review by antalszava — COMMENTED** [review]
Just one question and an observation: hovering over the sidebar items now makes the titles blue whereas it used to make them green. I think this is intentional, just double-checking that it is indeed so. :slightly_smiling_face: (it fits with the new logo amazing!)

**Comment by josh146** [comment]
> hovering over the sidebar items now makes the titles blue whereas it used to make them green.

Yep, this is intentional :)

File: doc/conf.py [hunk 2]
```diff
@@ -225,69 +223,15 @@
...
+html_theme = "pennylane"
 
 # Xanadu theme options (see theme.conf for more information).
 html_theme_options = {
-    "navbar_logo_path": "_static/logo.png",
-    "navbar_wordmark_path": "_static/pennylane.svg",
-    # Specifying #19b37b is more correct but does not match the other PL websites.
-    "navbar_logo_colour": "#2d7c7f",
-    "navbar_home_link": "https://pennylane.ai",
-    "navbar_left_links": [
```
Comment (antalszava) [inline doc/conf.py:237]: These seem to be removed for good (the new theme doesn't define them explicitly). That's something we'd like to do, right?

File: doc/requirements.txt [hunk 3]
```diff
@@ -32,4 +32,4 @@ torchvision==0.12.0; sys_platform == "darwin" and python_version == "3.10"
 jinja2==3.0.3
 retworkx==0.11.0
 networkx==2.6
-xanadu-sphinx-theme==0.1.0
+pennylane-sphinx-theme
```
Comment (antalszava) [inline doc/requirements.txt:35]: Can we pin here?
```suggestion
pennylane-sphinx-theme==0.3.1
```

**Review by antalszava — APPROVED** [review]
Looks good! :slightly_smiling_face: Thanks for the answers & pointer. Any blockers to pinning the version of the theme?

File: doc/conf.py (hunk 2 above)
Comment (josh146) [reply doc/conf.py:237]: The new theme does define them, in the `pennylane_sphinx_theme/__init__.py` :)

File: doc/requirements.txt (hunk 3 above)
Comment (josh146) [reply doc/requirements.txt:35]: I deliberately do not pin, as this allows us to update the navbar/footer in a single repo, and have it automatically propagate to all other PL repos 🙂 

It's not ideal, but if we pin in each repo, we are back in a situation where we need to still update 16 different repos just to change the navbar!

Comment (josh146) [reply doc/requirements.txt:35]:
```suggestion
# we do not pin the sphinx theme, to allow the
# navbar and header data to be updated at the source
pennylane-sphinx-theme
```

Comment (antalszava) [reply doc/requirements.txt:37]: I think that sounds very reasonable! Could we have a comment here in the requirements file about that?

We've started following the strict pinning approach in the requirements files, so good to document the reason for having no pin here.

**Pull request merged**